### PR TITLE
Enhanced Logging to be cached until the end "SUBMIT" plus bug fixes

### PR DESCRIPTION
**Pull request by Jeff-Stammler**
Now all logging is Cached until Submit, when we write them out.  Quitting the app in the middle won't leave half written files.  Sets us up for a match reset easily, and an undo button.

Fixed how we call/close activities using finish() so we don't create an endless stack (of pancakes?).

That fixed the screen flipping issue.

Added the team number/name to the Match page.

### CODE REVIEW

File: app/src/main/java/com/cpr3663/cpr_scouting_app/Settings.java [hunk 1]
```diff
@@ -221,7 +221,7 @@ public void onClick(View view) {
                 Globals.CurrentPrefTeamPos = spinner_PrefTeamPos.getSelectedItemPosition();
                 Globals.spe.putInt(Constants.SP_PREF_TEAM_POS, Globals.CurrentPrefTeamPos);
                 Globals.spe.apply();
-                Exit();
+                finish();
```
Comment (AdrielStammler) [inline app/src/main/java/com/cpr3663/cpr_scouting_app/Settings.java:224]: Cool makes sense why and how you are doing this. But is there any reason to not just delete the function "Exit"? if not do so.

**Review by AdrielStammler — CHANGES_REQUESTED** [review]

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -24,6 +24,7 @@
         </activity>
         <activity
             android:name=".Settings"
+            android:parentActivityName=".AppLaunch"
```
[the remark this answers is not in the document]
Comment (Jeff-Stammler) [reply app/src/main/AndroidManifest.xml:27]: When you "Finish" a child activity android goes back to the parent.  Yep. Cool!

File: app/src/main/java/com/cpr3663/cpr_scouting_app/Settings.java (hunk 1 above)
Comment (Jeff-Stammler) [reply app/src/main/java/com/cpr3663/cpr_scouting_app/Settings.java:224]: Exit is deleted, hence it's in red.

Is your asking why change it, themes be sure finish is more "expected" in android and runs any cleanup, etc.

Or did you mean something else?

Comment (AdrielStammler) [reply app/src/main/java/com/cpr3663/cpr_scouting_app/Settings.java:224]: Not that the function below.

Comment (Jeff-Stammler) [reply app/src/main/java/com/cpr3663/cpr_scouting_app/Settings.java:224]: done.

**Review by AdrielStammler — APPROVED** [review]

**Pull request merged**